ios/src/TangramMap/TGMapViewController.h

```diff
@@ -0,0 +1,731 @@
+#import "TGMarkerPickResult.h"
+
+/**
+ Description for a <a href="https://mapzen.com/documentation/tangram/Cameras-Overview/#camera-types">
+ Tangram camera type</a>.
+ */
+typedef NS_ENUM(NSInteger, TGCameraType) {
+    /// Type for a <a href="https://mapzen.com/documentation/tangram/Cameras-Overview/#perspective-camera">perspective camera</a>
+    TGCameraTypePerspective = 0,
+    /// Type for an <a href="https://mapzen.com/documentation/tangram/Cameras-Overview/#isometric-camera">isometric camera</a>
+    TGCameraTypeIsometric,
+    /// Type for a <a href="https://mapzen.com/documentation/tangram/Cameras-Overview/#flat-camera">flat camera</a>
+    TGCameraTypeFlat,
+};
+
+/**
+ These defines describe an ease type function to be used for camera or other transition animation.
+ The function is being used to interpolate between the start and end position of the animation.
+ */
+typedef NS_ENUM(NSInteger, TGEaseType) {
+    /// Linear ease type `f(t) = t`
+    TGEaseTypeLinear = 0,
+    /// Cube ease type `f(t) = (-2 * t + 3) * t * t`
+    TGEaseTypeCubic,
+    /// Quint ease type `f(t) = (6 * t * t - 15 * t + 10) * t * t * t`
+    TGEaseTypeQuint,
+    /// Sine ease type `f(t) = 0.5 - 0.5 * cos(PI * t)`
+    TGEaseTypeSine,
+};
+
+/// Debug flags to render additional information about various map components
+typedef NS_ENUM(NSInteger, TGDebugFlag) {
+    /// While on, the set of tiles currently being drawn will not update to match the view
+    TGDebugFlagFreezeTiles = 0,
+    /// Apply a color change to every other zoom level to visualize proxy tile behavior
+    TGDebugFlagProxyColors,
+    /// Draw tile boundaries
+    TGDebugFlagTileBounds,
+    /// Draw tile infos (tile coordinates)
+    TGDebugFlagTileInfos,
+    /// Draw label bounding boxes and collision grid
+    TGDebugFlagLabels,
+    /// Draw tangram infos (framerate, debug log...)
+    TGDebugFlagTangramInfos,
+    /// Draw all labels (including labels being occluded)
+    TGDebugFlagDrawAllLabels,
+    /// Draw tangram frame graph stats
+    TGDebugFlagTangramStats,
+    /// Draw feature selection framebuffer
+    TGDebugFlagSelectionBuffer,
+};
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ A gesture recognizer delegate that can be implemented to receive gesture events from the map view.
+ The map view will first check whether a gestureDelegate is available, then check whether it responds
+ to any `shouldRecognize*` method:
+
+    - If the `TGRecognizerDelegate` responds to `shouldRecognize`, the map view calls it before proceeding with
+    its default implementation based on the resturn status of `shouldRecognize`.
+
+    - If the `TGRecognizerDelegate` doesn't respond to `shouldRecognize*`, the map view will proceed with
+    calling default gesture handling code.
+
+ Finally, the map view checks whether the `TGRecognizerDelegate` implements `didRecognize*` and call this
+ method if so.
+
+ @note Those methods are all **optional**.
+ */
+@protocol TGRecognizerDelegate <NSObject>
```

Comment:
Code organization thought (my OCD kicking in) for these methods:

1) Put all the *shouldRecognize*  functions together, and all the *didRecognize* functions together beneath them, since that's the order they happen in.

OR

2) Put the *shouldRecognizeX* function and the corresponding *didRecognizeX* function together in blocks, to indicate the pairing. I can't remember without looking at the code but I'm pretty sure this is how the implementation file is organized.